taf/tests/test_updater/test_update_library/test_validation_and_sync.py

```diff
@@ -0,0 +1,294 @@
+        kwargs={"origin_auth_repo": origin_auth_repo, "client_dir": client_dir},
+    )
+    setup_manager.execute_tasks()
+
+    # Run the updater to update repositories
+    update_and_check_commit_shas(
+        OperationType.UPDATE,
+        origin_auth_repo,
+        client_dir,
+    )
+    check_repo_sync(client_dir, origin_auth_repo)
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [
+                {"name": "target1", "allow_unauthenticated_commits": True},
+                {"name": "target2", "allow_unauthenticated_commits": True},
+            ],
+        },
+    ],
+    indirect=True,
+)
+def test_auth_repo_not_in_sync_partial(origin_auth_repo, client_dir):
+    """
+    Clients auth repo has been manually updated, just like in 1.
+    The update should result in a partial update.
+    Clients auth repo should be reverted to the last valid commit,
+    last validated commits set to that value,
+    target repositories updated up to commits listed in auth repo's last validated commits
+    """
+    clone_repositories(
+        origin_auth_repo,
+        client_dir,
+    )
+
+    # Load the target repositories
+    target_repos = list(load_target_repositories(origin_auth_repo).values())
+    original_commits = {repo.name: repo.head_commit_sha() for repo in target_repos}
+
+    setup_manager = SetupManager(origin_auth_repo)
+    setup_manager.add_task(add_valid_unauthenticated_commits)
+    setup_manager.execute_tasks()
+
+    new_commits = {repo.name: repo.head_commit_sha() for repo in target_repos}
+
+    for repo_name in original_commits:
+        assert original_commits[repo_name] != new_commits[repo_name]
+
+    setup_manager.add_task(sync_auth_repo)
+    setup_manager.execute_tasks()
+
+    update_and_check_commit_shas(OperationType.UPDATE, origin_auth_repo, client_dir)
+    check_repo_sync(client_dir, origin_auth_repo)
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [
+                {"name": "target1", "allow_unauthenticated_commits": True},
+                {"name": "target2", "allow_unauthenticated_commits": True},
+            ],
+        },
+    ],
+    indirect=True,
+)
+def test_target_repo_not_in_sync_partial(origin_auth_repo, client_dir):
+    """
+    Same as 2, but target repositories contain invalid commits following commits which were manually pulled using git pull.
+    Expect all repositories to be updated up to the last validated commits and invalid commits to be removed
+    """
+    clone_repositories(
+        origin_auth_repo,
+        client_dir,
+    )
+    target_repos = list(load_target_repositories(origin_auth_repo).values())
+    original_commits = {repo.name: repo.head_commit_sha() for repo in target_repos}
+
+    setup_manager = SetupManager(origin_auth_repo)
+    setup_manager.add_task(add_valid_unauthenticated_commits)
+    setup_manager.execute_tasks()
```

Comment:
Add valid authenticates target commits following unauthenticated ones. That will result in an invalid update. Should we combine the two to make it easier to set up this invalid state?